gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagProcArbiterDecorator.java

```diff
@@ -0,0 +1,140 @@
+ */
+public class DagProcArbiterDecorator extends InstrumentedLeaseArbiterDecorator {
+  private final DagProcReminderScheduler dagProcReminderScheduler;
+
+  @Inject
+  public DagProcArbiterDecorator(Config config, Optional<MultiActiveLeaseArbiter> leaseDeterminationStore,
+      DagProcReminderScheduler dagProcReminderScheduler) throws IOException {
+    super(config, leaseDeterminationStore.get(), String.valueOf(DagProcArbiterDecorator.class));
+    this.dagProcReminderScheduler = dagProcReminderScheduler;
+  }
+
+  /**
+   * This method is used by the multi-active scheduler and multi-active execution classes (DagTaskStream) to attempt a
+   * lease for a particular job event and return the status of the attempt.
+   * @param flowAction
+   * @param eventTimeMillis
+   * @param isReminderEvent
+   * @param skipFlowExecutionIdReplacement
+   * @return
+   */
+  @Override
+  public MultiActiveLeaseArbiter.LeaseAttemptStatus tryAcquireLease(DagActionStore.DagAction flowAction, long eventTimeMillis,
+      boolean isReminderEvent, boolean skipFlowExecutionIdReplacement) {
+    try {
+      MultiActiveLeaseArbiter.LeaseAttemptStatus leaseAttemptStatus = this.tryAcquireLease(flowAction, eventTimeMillis, isReminderEvent, skipFlowExecutionIdReplacement);
+      if (leaseAttemptStatus instanceof MultiActiveLeaseArbiter.LeasedToAnotherStatus) {
+        scheduleReminderForEvent((MultiActiveLeaseArbiter.LeasedToAnotherStatus) leaseAttemptStatus, eventTimeMillis);
+        return leaseAttemptStatus;
+      }
+    /* Otherwise leaseAttemptStatus instanceof {@link MultiActiveLeaseArbiter.LeaseObtainedStatus/NoLongerLeasingStatus}
+      and no need to do anything
+     */
+      return leaseAttemptStatus;
+    } catch (SchedulerException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  protected void scheduleReminderForEvent(MultiActiveLeaseArbiter.LeasedToAnotherStatus leaseStatus, long triggerEventTimeMillis)
+      throws SchedulerException {
+    // TODO: determine which ts to use
+    dagProcReminderScheduler.scheduleReminderJob(leaseStatus.getFlowAction(), leaseStatus.getEventTimeMillis());
+
+  }
+
+  @Slf4j
+  public class ReminderJob implements Job {
+    public static final String FLOW_ACTION_TYPE_KEY = "flow.actionType";
+    public static final String DAG_TASK_STREAM = "dag.taskStream";
+
+    @Override
+    public void execute(JobExecutionContext context) throws JobExecutionException {
+      // Get properties from the trigger to create a dagAction
+      String flowName = context.getTrigger().getJobDataMap().getString(ConfigurationKeys.FLOW_NAME_KEY);
+      String flowGroup = context.getTrigger().getJobDataMap().getString(ConfigurationKeys.FLOW_GROUP_KEY);
+      String jobName = context.getTrigger().getJobDataMap().getString(ConfigurationKeys.JOB_NAME_KEY);
+      String flowId = context.getTrigger().getJobDataMap().getString(ConfigurationKeys.FLOW_EXECUTION_ID_KEY);
```

Comment:
the repetitive `context.getTrigger().getJobDataMap().` adds clutter.  could we just define a locally-scoped `dataMap` at the top?